panic happens when configured

im trying to configure the config according to the following docs
this documentation https://github.com/devopsfaith/krakend-documentation/blob/85d586a05a36a24abf894677c3521eeeff1d41b9/endpoints/security.md 

```
        "github_com/devopsfaith/krakend-httpsecure": {
            "allowed_hosts": [],
            "ssl_proxy_headers": {},
            "sts_seconds": "300",
            "frame_deny": false,
            "sts_include_subdomains": false
        },
```
with this in mind i configured the config like above which results in a panic. 

see the trace 
```
panic: interface conversion: interface {} is string, not int

goroutine 1 [running]:
github.com/devopsfaith/krakend-ce/vendor/github.com/devopsfaith/krakend-httpsecure.getInt64(...)
        /go/src/github.com/devopsfaith/krakend-ce/vendor/github.com/devopsfaith/krakend-httpsecure/httpsecure.go:74
github.com/devopsfaith/krakend-ce/vendor/github.com/devopsfaith/krakend-httpsecure.ConfigGetter(0xc0007019b0, 0xc00071a800, 0x5)
        /go/src/github.com/devopsfaith/krakend-ce/vendor/github.com/devopsfaith/krakend-httpsecure/httpsecure.go:32 +0x790
github.com/devopsfaith/krakend-ce/vendor/github.com/devopsfaith/krakend-httpsecure/gin.Register(0xc0007019b0, 0xc00034c6c0, 0x1, 0x1)
```


what i tried voor the `sts_seconds` is  
- integer  (300)
- string ("300") 

they both raise a panic

--- a/httpsecure.go
+++ b/httpsecure.go
@@ -51,7 +51,9 @@ func getStrings(data map[string]interface{}, key string, v *[]string) {
 	if vs, ok := data[key]; ok {
 		result := []string{}
 		for _, v := range vs.([]interface{}) {
-			result = append(result, v.(string))
+			if s, ok := v.(string); ok {
+				result = append(result, s)
+			}
 		}
 		*v = result
 	}
@@ -60,17 +62,29 @@ func getStrings(data map[string]interface{}, key string, v *[]string) {
 func getString(data map[string]interface{}, key string, v *string) {
 	if val, ok := data[key]; ok {
 		*v = val.(string)
+		if s, ok := val.(string); ok {
+			*v = s
+		}
 	}
 }
 
 func getBool(data map[string]interface{}, key string, v *bool) {
 	if val, ok := data[key]; ok {
-		*v = val.(bool)
+		if b, ok := val.(bool); ok {
+			*v = b
+		}
 	}
 }
 
 func getInt64(data map[string]interface{}, key string, v *int64) {
 	if val, ok := data[key]; ok {
-		*v = int64(val.(int))
+		switch i := val.(type) {
+		case int64:
+			*v = i
+		case int:
+			*v = int64(i)
+		case float64:
+			*v = int64(i)
+		}
 	}
 }

--- a/httpsecure_test.go
+++ b/httpsecure_test.go
@@ -1,21 +1,46 @@
 package httpsecure
 
 import (
+	"encoding/json"
 	"fmt"
-	"testing"
 
 	"github.com/devopsfaith/krakend/config"
 )
 
-func TestConfigGetter(t *testing.T) {
+func ExampleConfigGetter() {
 	cfg := ConfigGetter(config.ExtraConfig{
 		Namespace: map[string]interface{}{
 			"allowed_hosts":      []interface{}{"host1"},
 			"host_proxy_headers": []interface{}{"x-custom-header"},
-			"sts_seconds":        10,
+			"sts_seconds":        10.0,
 			"ssl_redirect":       true,
 			"ssl_host":           "secure.example.com",
 		},
 	})
 	fmt.Println(cfg)
+
+	// output:
+	// {false false false false false false true false false false false       secure.example.com [host1] [x-custom-header] <nil> map[] 10 }
+}
+
+func ExampleConfigGetter_fromParsedData() {
+	sample := `{
+            "allowed_hosts": ["host1"],
+            "ssl_proxy_headers": {},
+            "sts_seconds": 300,
+            "frame_deny": true,
+            "sts_include_subdomains": true
+        }`
+	parsedCfg := map[string]interface{}{}
+
+	if err := json.Unmarshal([]byte(sample), &parsedCfg); err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	cfg := ConfigGetter(config.ExtraConfig{Namespace: parsedCfg})
+	fmt.Println(cfg)
+
+	// output:
+	// {false false false true false false false false false true false        [host1] [] <nil> map[] 300 }
 }

